Refactor functional tests with test helpers

kajinamit opened the pull request:
This refactors the existing functional tests using the helpers added to lib-common.

Also functional tests for webhook are added so that we can ensure default contaimer image names are properly assigned.

gibizer commented:
At some point we might want to make this configurable but we have the TODO there anyhow to fine tune them so this looks OK.
/lgtm

kajinamit committed:
Refactor function tests with test helpers

This refactors the existing functional tests using the helpers added
to lib-common.

kajinamit committed:
Add functional tests for PlacementAPI webhook

This introduces the functional tests for PlacementAPI webhook to ensure
default container images are set properly.

gibizer reviewed (approved):
Thanks Takashi!
Looks good to me and it is a test only change so it is safe to land even close to the dev preview
